src/concrete/ml/quantization/base_quantized_op.py

```diff
@@ -977,11 +977,12 @@ def cnp_round(
         n_bits = None
         exactness = fhe.Exactness.EXACT
 
-        if self.rounding_threshold_bits is not None:
-            # mypy
-            assert isinstance(self.rounding_threshold_bits, dict)
-            n_bits = self.rounding_threshold_bits.get("n_bits")
+        if isinstance(self.rounding_threshold_bits, dict):
+            n_bits = self.rounding_threshold_bits.get("n_bits", None)
             exactness = self.rounding_threshold_bits.get("method", exactness)
+        # PoT is replacing inplace the rounding_threshold_bits to an int
+        elif isinstance(self.rounding_threshold_bits, int):
```

Comment:
this is going to create some issues with typing I think if we are allowing `rounding_threshold_bits` to be an int

Reply:
No it should be ok. But PoT modifying `self.rounding_threshold_bits` on the fly is a bit annoying to handle. A small refacto is probably required.